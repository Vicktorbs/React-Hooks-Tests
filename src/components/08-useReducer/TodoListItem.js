import React from 'react'
import PropTypes from 'prop-types'

export const TodoListItem = ({todo, index, handleToggle, handleDelete}) => {
    return (
        <>
            <p className={ `${ todo.done && 'complete' }` } onClick={ () => handleToggle(todo.id) }>
                { index + 1 }. { todo.desc }
            </p>
            <button className="btn btn-danger" onClick={ () => { handleDelete(todo.id) } }>
                Borrar
            </button>
        </>
    )
}
TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired, 
    index: PropTypes.number.isRequired, 
    handleToggle: PropTypes.func.isRequired, 
    handleDelete: PropTypes.func.isRequired
}