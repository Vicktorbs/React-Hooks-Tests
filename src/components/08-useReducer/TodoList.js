import React from 'react'
import { TodoListItem } from './TodoListItem'
import PropTypes from 'prop-types'

export const TodoList = ({todos, handleDelete, handleToggle}) => {
    
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map((todo, i) => (
                    <li className="list-group-item" key={ todo.id }>
                        <TodoListItem todo={ todo } index={ i } handleToggle={ handleToggle } handleDelete={ handleDelete } />
                        {/* <p className={ `${ todo.done && 'complete' }` } onClick={ () => handleToggle(todo.id) }>{ i + 1 }. { todo.desc }</p>
                        <button className="btn btn-danger" onClick={ () => { handleDelete(todo.id) } }>
                            Borrar
                        </button> */}
                    </li>
                ))
            }
        </ul>
    )
}
TodoList.propTypes = {
    todos: PropTypes.array.isRequired, 
    handleToggle: PropTypes.func.isRequired, 
    handleDelete: PropTypes.func.isRequired
}