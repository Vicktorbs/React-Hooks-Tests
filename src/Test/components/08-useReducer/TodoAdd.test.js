import { shallow } from "enzyme"
import { TodoAdd } from "../../../components/08-useReducer/TodoAdd"

describe('Test on component <TodoAdd />', () => {
    
    const handleAddTodo = jest.fn();
    const wrapper = shallow(<TodoAdd 
        handleAddTodo={ handleAddTodo }
    />)
    test('should show itself correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should not call handleAddTodo', () => {
        const submitForm = wrapper.find('form').prop('onSubmit');
        submitForm({ preventDefault(){} });
        expect(handleAddTodo).toHaveBeenCalledTimes(0);
    })
    
    test('should call the function handleAddTodo', () => {
        const value = 'Lear how to stay there';
        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        });

        const submitForm = wrapper.find('form').prop('onSubmit');
        submitForm({ preventDefault(){} });
        expect(handleAddTodo).toHaveBeenCalledTimes(1);
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
        expect(handleAddTodo).toHaveBeenCalledWith({
            desc: "Lear how to stay there",
            done: false,
            id: expect.any(Number),
        });
        expect(wrapper.find('input').prop('value')).toBe('')
    })
    
})
