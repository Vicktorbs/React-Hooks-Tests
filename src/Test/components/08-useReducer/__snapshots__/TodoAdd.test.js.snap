// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on component <TodoAdd /> should show itself correctly 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-success m-2 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
